HLTrigger/JetMET/src/HLTMETCleanerUsingJetID.cc

```diff
@@ -97,10 +93,10 @@ void HLTMETCleanerUsingJetID::produce(edm::Event& iEvent, const edm::EventSetup&
         double mex_diff = mex_goodJets - mex_jets;
         double mey_diff = mey_goodJets - mey_jets;
         //double sumet_diff = sumet_goodJets - sumet_jets;  // cannot set sumet...
-        reco::Candidate::LorentzVector p4_diff(mex_diff, mey_diff, 0, sqrt(mex_diff*mex_diff + mey_diff*mey_diff));
+        reco::Candidate::LorentzVector p4_diff(met->front().px()+mex_diff, mey_diff+met->front().py(), 0, sqrt((met->front().px()+mex_diff)*(met->front().px() +mex_diff)+(met->front().py()+mey_diff)*(met->front().py() +mey_diff)));
 
-        reco::CaloMET cleanmet = met->front();
-        cleanmet.setP4(cleanmet.p4() + p4_diff);
+        reco::CaloMET cleanmet ;
+        cleanmet.setP4(p4_diff);
```

Comment:
Hi Batool,

thanks for putting up the PR!
Two questions/suggestions came to my mind after taking a look at the proposed changes:
- what about renaming p4_diff to e.g. cleanmetp4 (L96/99)? With the proposed code changes, it has a different meaning than before. It would be good to change the name, then.
- Does the change from reco::CaloMET cleanmet = met->front(); to initializing a fresh object reco::CaloMET cleanmet ; before setting the p4 loose any information that might be useful to have? The way you propose is definitely cleaner, I am just wondering...

Cheers,
Henning